packages/eslint-plugin/src/util/explicitReturnTypeUtils.ts

```diff
@@ -76,7 +76,9 @@ function getReporLoc(
  * const x: Foo = ...
  * ```
  */
-function isVariableDeclaratorWithTypeAnnotation(node: TSESTree.Node): boolean {
+function isVariableDeclaratorWithTypeAnnotation(
+  node: TSESTree.Node,
+): node is TSESTree.VariableDeclarator {
```

Comment:
this is not correct, negated types does not include `VariableDeclarator`

Reply:
what do you mean by this, sorry?

The function checks `node.type === AST_NODE_TYPES.VariableDeclarator`, so this is the correct guard.
I just didn't bother doing the `& { id: TSESTree.Identifier & { typeAnnotation: TSESTree.Identifier['typeAnnotation'] }`